src/runtime/internal/dom.ts

```diff
@@ -302,7 +302,10 @@ export function set_attributes(node: Element & ElementCSSInlineStyle, attributes
 		} else if (key === '__value') {
 			(node as any).value = node[key] = attributes[key];
 		} else if (descriptors[key] && descriptors[key].set) {
-			node[key] = attributes[key];
+			if (node[key] !== attributes[key]) {
+				node[key] = attributes[key];
+				if (node[key] !== attributes[key]) attr(node, key, attributes[key]);
```

Comment:
Isn't this already checked above? That if condition would never resolve to true because they are set to the same value the line above
```suggestion
				attr(node, key, attributes[key]);
```

Reply:
updated!